### ISSUE TITLE
XGBoost Benchmark cannot properly read config as a dict

### ISSUE THREAD

**Issue by NeoChaos12** [automl/HPOlib3#8]
When I tried to use the default configuration provided by the benchmark (local, non-containerized version) itself as a dict in a function call to objective_function(), it failed with the following error:

> Traceback (most recent call last):
>   File "/home/bansala/miniconda3/envs/hpolib3/lib/python3.8/site-packages/hpolib/abstract_benchmark.py", line 101, in wrapper
>     config_dict = {k: configuration[i] for (i, k) in enumerate(self.configuration_space)}
>   File "/home/bansala/miniconda3/envs/hpolib3/lib/python3.8/site-packages/hpolib/abstract_benchmark.py", line 101, in <dictcomp>
>     config_dict = {k: configuration[i] for (i, k) in enumerate(self.configuration_space)}
> KeyError: 0
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/home/bansala/xgboostbenchmarkmwe.py", line 24, in <module>
>     print(str(b.objective_function(test_config, n_estimators=n_estimators, subsample=subsample_ratio)))
>   File "/home/bansala/miniconda3/envs/hpolib3/lib/python3.8/site-packages/hpolib/abstract_benchmark.py", line 104, in wrapper
>     raise Exception('Error during the conversion of the provided into a ConfigSpace.Configuration'
> Exception: Error during the conversion of the provided into a ConfigSpace.Configuration object

A minimum working example of the code is:

```
import numpy as np
from hpolib.benchmarks.ml.xgboost_benchmark import XGBoostBenchmark as Benchmark

task_id = 167083
my_rng = np.random.RandomState(0)

b = Benchmark(task_id=task_id)

n_estimators = 128
subsample_ratio = 1.

cs = b.get_configuration_space()
test_config = cs.get_default_configuration().get_dictionary()

print(str(b.objective_function(test_config, n_estimators=n_estimators, subsample=subsample_ratio)))
```

The same behaviour was observed when a dict was manually constructed.

**Comment by PhMueller** [comment]
Hey,

Thanks for your feedback. I'll investigate the error as soon as possible. But I think it is due to some change in the api of the benchmark. Since this repository is still under  heavy development, the uploaded conatiner isnt up to date. 

As you already pointed out, the local version should work. Please use it until we were able to upload a new container.

Thanks, i'll keep you informed.

**Comment by PhMueller** [comment]
Should be fixed with commit #5.
